### PR TITLE
cleanup_before: clean `/Library/Frameworks` directory

### PR DESCRIPTION
**Pull request by carlocab**
These directories contain headers and libraries that we don't want to
use, so let's get rid of them when building on a GitHub macOS runner.

The Mono framework in particular contains Gettext headers that I've seen
mess up CMake builds in the past.

### CODE REVIEW

File: lib/tests/cleanup_before.rb [hunk 1]
```diff
@@ -24,6 +24,20 @@ def run!(args:)
...
+              Mono.framework
+              PluginManager.framework
+              Python.framework
+              R.framework
+              Xamarin.Android.framework
+              Xamarin.Mac.framework
+              Xamarin.iOS.framework
+            ].each do |framework|
+              path = frameworks_dir/framework
+              test "sudo", "mv", path, HOMEBREW_TEMP if path.exist?
```
Comment (MikeMcQuaid) [inline lib/tests/cleanup_before.rb:39]: Can this use `delete_or_move` instead?

Comment (carlocab) [reply lib/tests/cleanup_before.rb:39]: Needs `sudo`, unfortunately. Would you rather we refactor `delete_or_move` to use `sudo` when requested?

Comment (MikeMcQuaid) [reply lib/tests/cleanup_before.rb:39]: @carlocab Might make sense but: up to you, no strong feelings other way!

Comment (carlocab) [reply lib/tests/cleanup_before.rb]: Done!

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**